### PR TITLE
[_]: Test/e2e

### PR DESCRIPTION
**Pull request by JoanVicens**
Added e2e tests with a github workflow

### CODE REVIEW

File: tests/e2e/storage.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+  await server.database.query('DELETE FROM deleted_files WHERE user_id = (:userId)', {
+    replacements: { userId },
+  });
+};
+
+const createFolder = async (body: any, authToken: string): Promise<request.Response> =>
+  await request(app).post('/api/storage/folder').set('Authorization', `Bearer ${authToken}`).send(body);
+
+const createFileOnFolder = async (folderId: number, name: string, authToken: string): Promise<request.Response> =>
```
Comment (fabioespinosa) [inline tests/e2e/storage.e2e-spec.ts:38]: I also like to do these re-usable functions for tests, makes it much easier to read

File: tests/e2e/storage.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+import { FileModel } from '../../src/app/models/file';
+import { FolderModel } from '../../src/app/models/folder';
+import sequelize from 'sequelize';
+const { Op } = sequelize;
+const server = require('../../src/app');
+const app = server.express;
+
+const deleteAllUserFilesFromDatabase = async (userId: number): Promise<void> => {
+  const files = await server.models.file.findAll({ where: { user_id: userId } });
+  await files.forEach((file: FileModel) => file.destroy());
```
Comment (fabioespinosa) [inline tests/e2e/storage.e2e-spec.ts:16]: I'm not sure you can do await on forEach [1]

perhaps doing a 

for(const file of files){
 await file.destroy()
}

might work?

[1]: https://stackoverflow.com/a/37576787/4619199

File: tests/e2e/storage.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+};
+
+const deleteAllUserFoldersFromDatabase = async (userId: number): Promise<void> => {
+  const folders = await server.models.folder.findAll({ where: { user_id: userId } });
+  await folders.forEach((folder: FolderModel) => folder.destroy());
```
Comment (fabioespinosa) [inline tests/e2e/storage.e2e-spec.ts:21]: same here

File: tests/e2e/storage.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+};
+
+const clearUserDrive = async (userId: number, rootFolderId: number): Promise<void> => {
+  await deleteAllUserFilesFromDatabase(userId);
+
+  const folders = await server.models.folder.findAll({ where: { user_id: userId, id: { [Op.not]: rootFolderId } } });
+  await folders.map((folder: FileModel) => folder.destroy());
```
Comment (fabioespinosa) [inline tests/e2e/storage.e2e-spec.ts:28]: Here you can use:

`await Promise.all(folders.map((folder: FileModel) => folder.destroy()));`

Comment (JoanVicens) [reply tests/e2e/storage.e2e-spec.ts]: 👌🏼 changed!

File: tests/e2e/storage.e2e-spec.ts (hunk 3 above)
Comment (JoanVicens) [reply tests/e2e/storage.e2e-spec.ts]: 👌🏼 changed!

File: tests/e2e/utils.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+export async function delay(seconds: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    },
+    seconds * 1000);
+  });
+}
```
Comment (fabioespinosa) [inline tests/e2e/utils.ts:60]: end of file line

Comment (JoanVicens) [reply tests/e2e/utils.ts]: 👌🏼 changed!

**Review by fabioespinosa — APPROVED** [review]

**Pull request merged**